backend/apps/progress/tasks.py

```
@@ -177,3 +165,14 @@ def analyze_submission_plagiarism(submission_id: int):
             # Stop after first match as tests expect a single report
             break
+
+
+def update_leaderboard_task(user_id, username, xp_delta):
+    """
+    Background task to update Redis leaderboard and broadcast WebSocket event.
+    """
+    try:
+        from apps.progress.services.leaderboard_service import LeaderboardService
+        LeaderboardService.update_user_xp(user_id, username, xp_delta)
+    except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)